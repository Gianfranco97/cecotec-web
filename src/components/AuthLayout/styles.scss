.auth-page-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  background-color: #00262b;

  .auth-form {
    width: 400px;
    padding: 50px;
    border-radius: 15px;
    background-color: white;

    .auth-form-logo {
      text-align: center;
      margin-bottom: 30px;
    }
  }
}
