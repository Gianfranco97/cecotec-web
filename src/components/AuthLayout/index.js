import React from "react";
import PropTypes from "prop-types";
import logoIMG from "../../assets/img/cecotec-logo.jpg";

import "./styles.scss";

class AuthLayout extends React.Component {
  render() {
    const { children } = this.props;
    return (
      <div className="auth-page-container">
        <div className="auth-form">
          <div className="auth-form-logo">
            <img src={logoIMG} alt="cecotec" />
          </div>

          {children}
        </div>
      </div>
    );
  }
}

AuthLayout.propTypes = {
  children: PropTypes.element.isRequired,
};

export default AuthLayout;
