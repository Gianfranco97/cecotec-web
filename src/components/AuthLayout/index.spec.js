import React from "react";
import { shallow } from "enzyme";
import AuthLayout from "./index.js";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

it("The snapshot matches", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <AuthLayout>
          <div />
        </AuthLayout>
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(
    <AuthLayout>
      <div />
    </AuthLayout>
  );
});
