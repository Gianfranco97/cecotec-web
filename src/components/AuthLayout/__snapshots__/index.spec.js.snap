// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The snapshot matches 1`] = `
<div
  className="auth-page-container"
>
  <div
    className="auth-form"
  >
    <div
      className="auth-form-logo"
    >
      <img
        alt="cecotec"
        src="cecotec-logo.jpg"
      />
    </div>
    <div />
  </div>
</div>
`;
